test2/task2/console.c

```diff
@@ -0,0 +1,54 @@
+#include "console.h"
+#include "graph.h"
+#include <stdlib.h>
+
+static void freeTable(unsigned int*** const table, const size_t length)
+{
+    for (size_t i = 0; i < length; ++i)
+    {
+        free((*table)[i]);
+    }
+    free(*table);
+    *table = NULL;
+}
+
+static unsigned int** scanTable(const size_t rows, const size_t columns)
+{
+    unsigned int** table = (unsigned int**)malloc(rows * sizeof(unsigned int*));
+    if (table == NULL)
+    {
+        return NULL;
+    }
+    for (size_t i = 0; i < rows; ++i)
+    {
+        table[i] = (unsigned int**)malloc(columns * sizeof(unsigned int));
```

Comment:
```suggestion
        table[i] = (unsigned int*)malloc(columns * sizeof(unsigned int));
```
Иначе оно даже не компилируется, типы не сходятся. `table[i]` имеет тип `int*`